boa/src/builtins/function/arguments.rs

```diff
@@ -160,87 +161,83 @@ impl Arguments {
-                        };
-                        // 2. Let p be MakeArgSetter(name, env).
-                        // https://tc39.es/ecma262/#sec-makeargsetter
-                        let p = {
-                            // 2. Let setter be ! CreateBuiltinFunction(setterClosure, 1, "", « »).
-                            // 3. NOTE: setter is never directly accessible to ECMAScript code.
-                            // 4. Return setter.
-                            FunctionBuilder::closure_with_captures(
-                                context,
-                                // 1. Let setterClosure be a new Abstract Closure with parameters (value) that captures
-                                // name and env and performs the following steps when called:
-                                |_, args, captures, context| {
-                                    let value = args.get(0).cloned().unwrap_or_default();
-                                    // a. Return env.SetMutableBinding(name, value, false).
-                                    captures
-                                        .0
-                                        .set_mutable_binding(captures.1, value, false, context)
-                                        .map(|_| JsValue::Undefined)
-                                    // Ok(JsValue::Undefined)
-                                },
-                                (env.clone(), parameter_name),
-                            )
-                            .length(1)
-                            .name("")
-                            .build()
-                        };
-
-                        // 3. Perform map.[[DefineOwnProperty]](! ToString(𝔽(index)), PropertyDescriptor {
-                        // [[Set]]: p, [[Get]]: g, [[Enumerable]]: false, [[Configurable]]: true }).
-                        map.__define_own_property__(
-                            index.into(),
-                            PropertyDescriptor::builder()
-                                .set(p)
-                                .get(g)
-                                .enumerable(false)
-                                .configurable(true)
-                                .build(),
-                            context,
-                        )
-                        .expect("[[DefineOwnProperty]] must not fail per the spec");
-                    }
+        // Get the mapping of the environment binding index to the index of the object property.
+        // Parameters can be defined multiple times, so the mapping is not one to one.
+        let mut bindings = FxHashMap::default();
+        let mut property_index = 0;
+        'outer: for formal in formals {
+            for name in formal.names() {
+                if property_index >= len {
+                    break 'outer;
                 }
+                let binding_index = bindings.len() + 1;
+                let entry = bindings
+                    .entry(name)
```

Comment:
The entry API is very nice for this :) it might be useful elsewhere in Boa too. It wasn't there when many things in the compiler where developed :)